public/app/features/dashboard/state/reducers.ts

```diff
@@ -30,9 +31,13 @@ const dashboardSlice = createSlice({
     dashboardInitServices: (state) => {
       state.initPhase = DashboardInitPhase.Services;
     },
-    dashboardInitCompleted: (state, action: PayloadAction<DashboardModel>) => {
-      state.getModel = () => action.payload;
+    dashboardInitCompleted: (
+      state,
+      action: PayloadAction<{ dashboard: DashboardModel; original: DashboardDataDTO | null }>
+    ) => {
+      state.getModel = () => action.payload.dashboard;
       state.initPhase = DashboardInitPhase.Completed;
+      state.getOriginal = () => action.payload.original;
```

Comment:
Tried saving the JSON directly without a wrapper function but get this error:
![Screenshot 2023-08-28 at 8 53 57](https://github.com/grafana/grafana/assets/8878045/04e1b1c8-3678-4122-bf20-48f5613c7c72)

Reply:
ah, yes. I think we modify the object to remove all the properties that should not be included in the diff